conans/client/loader.py

```diff
@@ -70,8 +71,11 @@ def __init__(self, runner, settings, profile):
         @param cached_env_values: EnvValues object
         """
         self._runner = runner
+
+        assert isinstance(settings, Settings)
         settings.values = profile.settings_values
-        assert settings is None or isinstance(settings, Settings)
+        preprocess(settings, output)
```

Comment:
Why is the responsibility of the loader to mix the settings values with the settings?

`settings.values = profile.settings_values`

I agree with your observation but then let's extract it too. You need the values to preprocess the settings.